modules/sample-dropwizard/src/test/scala/core/Dropwizard/DropwizardRoundTripTest.scala

```diff
@@ -0,0 +1,90 @@
+package core.Dropwizard
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import examples.client.dropwizard.user.UserClient
+import examples.server.dropwizard.definitions.User
+import examples.server.dropwizard.user._
+import helpers.MockHelpers._
+import java.util
+import java.util.Optional
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+import org.asynchttpclient.{Request, Response}
+import org.mockito.{ArgumentMatchersSugar, MockitoSugar}
+import org.scalatest.concurrent.Waiters
+import org.scalatest.{FreeSpec, Matchers}
+import scala.compat.java8.FunctionConverters._
+
+class DropwizardRoundTripTest extends FreeSpec with Matchers with Waiters with MockitoSugar with ArgumentMatchersSugar {
+  private implicit val mapper = new ObjectMapper
+
+  "Test server" in {
+    val USERNAME = "foobar"
+
+    val (asyncResponse, serverFuture: CompletableFuture[GetUserByNameResponse]) = mockAsyncResponse[GetUserByNameResponse]
+
+    val resource = new UserResource(new UserHandler {
+      override def createUser(body: User): CompletionStage[CreateUserResponse] = ???
+      override def createUsersWithArrayInput(body: util.List[User]): CompletionStage[CreateUsersWithArrayInputResponse] = ???
+      override def createUsersWithListInput(body: util.List[User]): CompletionStage[CreateUsersWithListInputResponse] = ???
+      override def loginUser(username: String, password: String): CompletionStage[LoginUserResponse] = ???
+      override def logoutUser(): CompletionStage[LogoutUserResponse] = ???
+      override def updateUser(username: String, body: User): CompletionStage[UpdateUserResponse] = ???
+      override def deleteUser(username: String): CompletionStage[DeleteUserResponse] = ???
+
+      override def getUserByName(username: String): CompletionStage[GetUserByNameResponse] = {
+        username match {
+          case USERNAME =>
+            serverFuture.complete(new GetUserByNameResponse.Ok(User.builder()
```

Comment:
How are the ergonomics around this vs a dedicated `respond` parameter? It's discoverable because the method explicitly states that it needs one in the return type, but the repetition of the parameter name every time we build a response is less desirable.

Reply:
So the difference, just so I understand... with the current codegen, to return from a handler method you'd do
```java
return new GetUserByNameResponse.Ok(user);
```
But with how scala does, it, you'd get a `respond` param that has an `Ok` method (among others) so you'd do:
```java
return respond.Ok(user);
```
Yeah?

So... thinking about how to implement this, it's kinda icky in Java since there's no such thing as a companion object.  For Scala the `respond` object can look like:
```scala
object GetUserByNameResponse {
  def Ok(user: User): GetUserByNameResponseOk = GetUserByNameResponseOk(user)
}
```
But for Java... well, to get the same ergonomics we'd have to create a new class, and then pass an instance of it to the handler method:
```java
public class GetUserByNameResponseResponder {
    public GetUserByNameResponse.Ok Ok(final User user) {
        return new GetUserByNameResponse.Ok(user);
    }
}

// at site of handler call
handler.getUserByName(username, new GetUserByNameResponseResponder());

// then handler can do
response.Ok(user);
```
The naive idea would be to just create a class with static methods on it so we don't need an instance, but I don't see that as better; we can't actually pass an 'instance' of that class to the handler method since static methods in Java don't quite work the same way as methods on a companion object in Scala.

I do very much like the idea of having an instance/class/something right there front and center where you can see what you can/need to do to return various types of responses, vs. having to find the response class, inspect what inner subclasses exist, and then create instances via `new`.  That's not that discoverable.

How about something in the middle?  Providing a `respond` instance honestly feels weird to me in Java-land, so how about we do something in between:
```java
public abstract class GetUserByNameResponse {
    public static class Ok extends GetUserByNameResponse {
        private Ok(final User value) { ... }
        // field and getter method
    }

    public static class notFound extends GetUserByNameResponse {
        private NotFound() { ... }
    }

    public static Ok Ok(final User value) {
        return new Ok(value);
    }

    public static NotFound() {
        return new NotFound();
    }
}
```
There's still some discoverability required, but there's less of a burden.  The user sees that the handler method needs to return `CompletionStage<GetUserByNameResponse>`, so perhaps they start typing `GetUserByNameResponse.` and then their IDE shows an autocompletion for `Ok` and `NotFound`, and then it's obvious what they can/should do.

It's not perfect, because they might try to type `new GetUserByNameResponse`, for which the IDE will be like "nope".  But this is... kinda in the middle, and not weird?

Reply:
I really like this as a compromise.

Additionally, considering all the static methods on java.time and related Java 8+ stdlib members (as well as nio), this seems more idiomatic than other options.

Reply:
Ok, this is done.  Pretty happy with how it turned out.  I ended up actually doing what I think is a little nicer: for response types that don't have a response body, instead of a static creator method, I just have e.g.:
```
public static final BadRequest = new BadRequest();
```
So you don't even need to create an instance of the object; there's just a singleton.